src/main/scala/firrtl/Compiler.scala

```diff
@@ -33,71 +33,75 @@ import java.io.Writer
 import Utils._
 import firrtl.passes._
 
-trait Compiler extends LazyLogging {
-  def run(c: Circuit, w: Writer)
+
+// ===========================================
+//                 Annotations
+// -------------------------------------------
+case class AnnotationException(message: String) extends Exception(message)
+trait Named { def name: String }
+case class ModuleName(name: String) extends Named
+case class ComponentName(name: String, module: ModuleName) extends Named
+
```

Comment:
I wonder if this `Named` can be unified with `HasName` and `IsDeclaration` in the IR.scala. Using references to the actual Named IR node would actually have a smaller memory footprint (since we don't have to allocate an additional string, although I plan to fix this later through unique `Identifier` objects) and it would provide more power. Similar with the ModuleName.

This different approach would require creating new annotation objects when IR nodes change even if their name does not change though, this may be an undesirable overhead.